duckduckgo/Android
# Site permissions store module

Task/Issue URL: https://app.asana.com/0/0/1202659341576430/f

### Description

Implemented site-permissions-store module with Room db and shared preferences instances so we can set when to show site permissions prompt

### Steps to test this PR
_Scenario 1_
- Install from this branch
- Go to `www.mictest.cc`
- Tap on "Test my cam" button
- Allow permission
- Refresh the page
- [ ] Check rationale doesn't appear again for the same website and same tab

_Scenario 2_
- Open new tab
- Go to `www.mictest.cc`
- [ ] Check rationale appear as it is a new session (different tab)


### No UI changes
